### ISSUE TITLE
java.lang.UnsatisfiedLinkError: dlopen failed: library "libwhirlyglobemaply.so" not found

### ISSUE THREAD

**Issue by louhy** [mousebird/WhirlyGlobe#1528]
I've tried going through the tutorial and after working past some minor issues (in another issue created earlier), I think it's "nearly" working but I get the error in the subject line. 

I've checked the WhirlyGlobeMaply.aar I've included (which by the way I needed to place in HelloEarth/libs to get it to build, rather than HelloEarth/app/libs as the tutorial states - not sure if due to a mistake I made or the tutorial). Also I tried putting a 2nd copy of it inside HelloEarth/app/libs in *addition* to the top level libs folder, just in case. 

Opening the AAR it does look a little odd to me - there's a jni folder which contains arm/armeabi/x86 folders, which have .so files. The odd part is there's also 3 empty folders of the same name (x86, etc.) at the top level of the aar which don't contain any files at all. AAR came from here (all versions appear the same): 
https://mousebird-consulting-inc.github.io/WhirlyGlobe/builds/builds.html

Thinking this was the issue I tried copying the so files from the jni folder into their respective folders at the top level, but I'm still getting the same error in the subject. Anything else I can try out? 

I'm developing on Linux, in case that's a factor. I've also tried running on both an emulator and a real phone.

**Comment by louhy** [comment]
Dang.. I tried using this finished version of the tutorial and I get the same library not found error. I guess this might be something Linux (or Android Studio setup?) specific. 

https://github.com/mousebird/AndroidTutorialProject

**Comment by sjg-wdw** [comment]
Can you take a quick look, @TimSylvester-mousebird

**Comment by TimSylvester-mousebird** [comment]
Nobody's checked out the tutorials in a while, I'll take a look.

**Comment by louhy** [comment]
I'm starting to wonder if it's not the tutorial, but something local to my environment. Either the fact that I'm developing from Linux or something to do with my Android Studio / emulator config. Can't explain why the cloned tutorial project would fail exactly the same way... 

Pretty sure the tutorial does have a few minor updates to do though, in the other issue I created.

**Comment by louhy** [comment]
Just in case it matters, I'm trying to use the current version of Android Studio with a "Galaxy S21 API 31" virtual device.

**Comment by louhy** [comment]
Realized that the S21 'emulator' was one I created so I tried a Galaxy Nexus API 30. I still had to tinker with a lot of things but I did get that to run... wow this stuff sure is picky. If I can figure out what it is I did to get it working I'll post here, if anyone is interested.

**Comment by EULUSOY** [comment]
Hi; did you find a solution to the 'java.lang.UnsatisfiedLinkError: dlopen failed: library' problem?

**Comment by TimSylvester** [comment]
No, it would seem to be that the library is not present on the device, but I can't tell why.  

It could be related to #1558, make sure the architecture of your target device is the NDK filter list.
